packages/core-commands/src/admin-navigation-commands.js

```diff
@@ -3,9 +3,30 @@
  */
 import { useCommand } from '@wordpress/commands';
 import { __ } from '@wordpress/i18n';
-import { external, plus } from '@wordpress/icons';
+import { external, plus, symbol } from '@wordpress/icons';
+import { useSelect } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { addQueryArgs, getPath } from '@wordpress/url';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from './lock-unlock';
+
+const { useHistory } = unlock( routerPrivateApis );
 
 export function useAdminNavigationCommands() {
+	const history = useHistory();
+	const getSettings = useSelect(
+		( select ) => select( blockEditorStore ).getSettings
+	);
+	const settings = getSettings();
+	const isBlockTheme = settings.__unstableIsBlockBasedTheme;
```

Comment:
Is there a reason for accessing settings during the render? I think it's better to return data component needs from `mapSelect`.

Should `core-commands` depend on the `block-editor` package?

```js
const isBlockTheme = useSelect( ( select ) => {
	return select( blockEditorStore ).getSettings()
		.__unstableIsBlockBasedTheme;
}, [] );
```

Reply:
We could use a string literal for a store if we don't want to add a `block-editor` package to deps. We have few similar precedents in the core already.

https://github.com/WordPress/gutenberg/blob/51774efbba2d334b04683842533e4fef1777b70b/packages/block-library/src/calendar/edit.js#L58-L61

Reply:
> Is there a reason for accessing settings during the render? I think it's better to return data component needs from `mapSelect`.

Fixed.

Reply:
> Should core-commands depend on the block-editor package?

I don't have a strong opinion on this, but it seems like the existing commands are very block editor based, so not sure if there will be cases where it will be used independently of it. Using the string literal seems to still make it dependent on the store but in a slightly flakier way. I may be overlooking other issues with this dependency though, so happy to update it to just use the string if others can see future potential issues with it.

Reply:
> Using the string literal seems to still make it dependent on the store but in a slightly flakier way.

With a tweak to check for the presence of the block editor selectors I think that removes the "dependency". 

The user would get the enhanced behaviour going to the site editor if it is available. If the commands package was being used with a custom editor or something, they'd get the standard wp-admin behaviour.

Perhaps it would be better to err towards not adding a package dependency until we're 100% sure it's worth it?

Reply:
ok, makes sense, will update it.

Reply:
Done.